javaslang/src-gen/main/java/javaslang/control/Applicative.java

```diff
@@ -0,0 +1,437 @@
+/*     / \____  _    _  ____   ______  / \ ____  __    _______
+ *    /  /    \/ \  / \/    \ /  /\__\/  //    \/  \  //  /\__\   JΛVΛSLΛNG
+ *  _/  /  /\  \  \/  /  /\  \\__\\  \  //  /\  \ /\\/ \ /__\ \   Copyright 2014-2016 Javaslang, http://javaslang.io
+ * /___/\_/  \_/\____/\_/  \_/\__\/__/\__\_/  \_//  \__/\_____/   Licensed under the Apache License, Version 2.0
+ */
+package javaslang.control;
+
+/*-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-*\
+   G E N E R A T O R   C R A F T E D
+\*-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-*/
+
+import java.util.function.BiFunction;
+import java.util.function.Function;
+import javaslang.collection.List;
+import javaslang.concurrent.Future;
+
+import javaslang.Function3;
+import javaslang.Function4;
+import javaslang.Function5;
+import javaslang.Function6;
+import javaslang.Function7;
+import javaslang.Function8;
+import javaslang.control.Option;
+import javaslang.control.Try;
+import javaslang.concurrent.Future;
+
+/**
+ * Applicative helpers
+ *
+ * @author Daniel Dietrich
+ * @since 2.1.0
+ */
+public final class Applicative {
```

Comment:
You are correct. This is `liftM2` not `liftA2`, because we use flatMap. And yes, we would need `<*>` to get liftA2. So let's keep what we have.

But the empty chain is not a difference. It's the same thing with applicative.

    > liftA2 (+) Nothing (Just 6)
    Nothing

Similarly:

    (+) <$> Nothing <*> (Just 6)
    Nothing

My understanding is that except for degenerate cases, `liftA2` and `liftM2` will return the same results. In haskell there is a goal to reach for the least specific mechanism, therefore it would be strange to pick `liftM2` when you can use `liftA2`. So this operation that we perform here would be referred to as "applicative lifting" by an haskeller, I think. But yeah we're not in haskell, we implement `liftM2` in reality, let's call it monadic lift.